src/cow-react/modules/limitOrders/hooks/useLimitOrdersTradeState.ts

```diff
@@ -0,0 +1,39 @@
+import { useAtomValue } from 'jotai/utils'
+import { limitOrdersAtom } from '@cow/modules/limitOrders/state/limitOrdersAtom'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import tryParseCurrencyAmount from 'lib/utils/tryParseCurrencyAmount'
+import { useMemo } from 'react'
+import { useTokenBySymbolOrAddress } from '@cow/common/hooks/useTokenBySymbolOrAddress'
+
+export interface LimitOrdersTradeState {
+  readonly inputCurrency: Currency | null
+  readonly outputCurrency: Currency | null
+  readonly inputCurrencyAmount: CurrencyAmount<Currency> | null
+  readonly outputCurrencyAmount: CurrencyAmount<Currency> | null
+  readonly recipient: string | null
+  readonly deadline: number | null
+}
+
+export function useLimitOrdersTradeState(): LimitOrdersTradeState {
+  const state = useAtomValue(limitOrdersAtom)
+
+  const recipient = state.recipient
+  const deadline = state.deadline
+  const inputCurrency = useTokenBySymbolOrAddress(state.inputCurrencyId) || null
+  const outputCurrency = useTokenBySymbolOrAddress(state.outputCurrencyId) || null
```

Comment:
`useTokenBySymbolOrAddress` already returns Token | null, so you don't need `|| null` here, do you?